### DISCUSSION TITLE
Commit title truncation should adapt to the available space

### DISCUSSION THREAD

**Discussion by smfr** [community/community#20888] [Other Feature Feedback, Questions, & Ideas]
There are various places in the GitHub UI where commit titles are truncated awkwardly, impacting usability.

For example, when viewing a [list of commits](https://github.com/WebKit/WebKit/compare/ea74655e102a2a6e63a9c71a38873ddb03f5bf33...8cf4b764ace993a01053ddc9371980e77a5260bb):

<img width="1254" alt="Screen Shot 2022-07-19 at 10 03 02 AM" src="https://user-images.githubusercontent.com/20968/179808189-f3b417d9-efe4-463d-bfad-e0956d3eec71.png">

Note that I've given the content more space by making my window wider, but commit titles are broken at 80 characters. This makes it really hard to quickly scan a list of commits to look for something.

Another place this happens is when viewing a [single commit](https://github.com/WebKit/WebKit/commit/bb3989fba3794bb836da052046927fbd9a4d5232):

<img width="1162" alt="Screen Shot 2022-07-19 at 10 04 12 AM" src="https://user-images.githubusercontent.com/20968/179808419-64e13d9d-5a89-48c7-8371-e9a4010d15a4.png">

Lots of horizontal space available, but the title is still annoyingly broken (in a way that makes it impossible to copy the test path, in this case).

I realize that this is echoing the default `git` 80 char truncation, but why can't the GitHub UI can't show longer titles without truncation?

**Comment by DeepSideEffect** [comment]
Actually, I have a case where I can see my full PR commit title from the commits view at `https://github.com/[myUserName]/[myProjetName]/commits/main/`, with the PR number as a link -> (#n) with n for the PR number.

But from the activity view at `https://github.com/[myUserName]/[myProjetName]/activity?ref=main`, it's truncated exactly right after the opening parenthesis (showing 72 characters but not the link to the PR number), so no easy way to link to the actual pull request of this activity flagged as _Pull request merge_ 😢
